### PR TITLE
Add disable progress bar feature, edit docstrings

### PR DESCRIPTION
**Pull request by Cubostar**
### Description

Adds feature explained in #3259. Also edited the docstrings to mention the new `progress` bool.

### CODE REVIEW

File: changelog/3280.feature.rst [hunk 1]
```diff
@@ -0,0 +1 @@
+Add ability to disable progressbars when dowloading files using `sunpy.net.helioviewer.py` and edited docstrings to mention this feature.
```
Comment (nabobalis) [inline changelog/3280.feature.rst]:
```suggestion
Add ability to disable progress bars when downloading files using `~sunpy.net.helioviewer.HelioviewerClient` and edited docstrings to mention this feature.
```

File: sunpy/net/helioviewer.py [hunk 2]
```diff
@@ -19,7 +19,7 @@
 __all__ = ['HelioviewerClient']
 
 
-class HelioviewerClient:
+class HelioviewerClient(object):
```
Comment (nabobalis) [inline sunpy/net/helioviewer.py]:
```suggestion
class HelioviewerClient:
```

File: sunpy/net/helioviewer.py [hunk 3]
```diff
@@ -128,7 +128,7 @@ def get_closest_image(self, date, observatory=None, instrument=None,
 
         return response
 
-    def download_jp2(self, date, observatory=None, instrument=None, detector=None,
+    def download_jp2(self, date, progress=True, observatory=None, instrument=None, detector=None,
```
Comment (nabobalis) [inline sunpy/net/helioviewer.py]: Move this to before the directory keywod.

**Comment by Cubostar** [comment]
Should tests be edited to check if this feature works?

**Comment by nabobalis** [comment]
I'm not sure I know how you would test it. @Cadair do you know a way?

**Comment by Cadair** [comment]
Easiest way would be to run a download and use pytest to capture the stdout and see if there is a progress bar or not.

(I am like 90% sure you can do that with pytest somehow.)

**Comment by Cubostar** [comment]
I'll do that in `~sunpy.net.tests.test_helioviewer.py` then

**Comment by Cubostar** [comment]
I haven't been able to test if these new tests I wrote work. Right now I'm assuming that `capsys` is a fixture (?) from `pytest` and is not a parameter, but I'm not quite sure how pytest works to run these.

**Comment by Cubostar** [comment]
Interestingly, there wasn't an AssertError even though `test_progress_png` didn't set `progress=False`.

**Comment by Cubostar** [comment]
Don't worry, just a rebase. The duplicate commits are because I reset my developer environment after making this pull request, so the branch `progressbar-feat` was only tracking the remote branch `progressbar-feat` on GitHub. The rebase basically made my local branch the official `progressbar-feat`, hence the extra commits (I think).

**Comment by Cubostar** [comment]
Tests now work! 😄

**Comment by ayshih** [comment]
@Cubostar While the general approach to clean up a commit history is through rebasing, you can instead limit yourself to just cherry-picking, which is easier to walk you through.  You'd create a fresh branch (here, `progressbar-feat-new`) and then cherry-pick the commits that you want to keep:
```
git fetch upstream
git checkout --no-track -b progressbar-feat-new upstream/master
git cherry-pick c007692^..2659b50
git cherry-pick 8ac93ea
```
After you confirm that the new branch does in fact contain all of your intended changes, you can rename the branch locally – overwriting the old branch – and then force push the branch to your remote:
```
git branch -M progressbar-feat-new progressbar-feat
git push -f origin progressbar-feat
```

**Comment by Cubostar** [comment]
WOW everything is so clean! Tysm @ayshih this is amazing

**Comment by Cubostar** [comment]
It seems that the CircleCI tests fail if `HelioviewerClient` doesn't have parentheses. Is this okay with you guys?

**Comment by ayshih** [comment]
The CircleCI builds are currently failing for reasons that are totally unrelated to this PR, not because of the recent change that you made.  You don't need to worry about it.

**Comment by Cadair** [comment]
Thanks @Cubostar :)

**Pull request merged**